src/SerializerGenerator.php

```diff
@@ -184,18 +184,7 @@ private function generateCodeForFieldType(
         string $modelPropertyPath,
         array $stack
     ): string {
-        if ($type instanceof PropertyTypeArray) {
-            if ($type->getSubType() instanceof PropertyTypePrimitive) {
-                // for arrays of scalars, copy the field even when its an empty array
-                return $this->templating->renderArrayAssign($fieldPath, $modelPropertyPath);
-            }
-
-            // either array or hashmap with second param the type of values
-            // the index works the same whether its numeric or hashmap
-            return $this->generateCodeForArray($type, $apiVersion, $serializerGroups, $fieldPath, $modelPropertyPath, $stack);
-        }
-
-        switch ($type) {
+        switch (true) {
```

Comment:
please keep this at `$type` (it does not technically matter whats in the switch statement, but i find it more expressive to know this switch is about the type)